### PR TITLE
add standalone rcs (cribbed from @steel4me) updated to latest osiris

### PR DESCRIPTION
**Pull request by r3muxd**

### CODE REVIEW

**Comment by SwagsRGB** [comment]
why cant rcs be used with silent aim? i am new to this so sorry if i asked smthng stupid

**Comment by r3muxd** [comment]
@SwagsRGB it's not a stupid question
i'll add a comment to clarify
you can't control recoil manually if you're using silent aim because it's silent, so silent aim has its own RCS system

**Comment by dayz2000** [comment]
work well

**Comment by SwagsRGB** [comment]
where is the rcs option? I cloned your repository entirely instead of merging the pull

**Comment by dayz2000** [comment]
> where is the rcs option? I cloned your repository entirely instead of merging the pull

Aimbot

**Comment by SwagsRGB** [comment]
works like a charm mate. nice job. But, sometimes it gets bugged if rcs and triggerbot is enabled together. however its quite rare ive just experienced it 2 times. All in all great job man.

**Comment by r3muxd** [comment]
fixed in latest ver

**Comment by r3muxd** [comment]
yees

**Comment by SwagsRGB** [comment]
hey i just wanna know is there no way RCS can be used with silent aim? I mean i use silent aim in my config but when i decide to spray i cant control the recoil. I know if rcs is coupled with silent aim rcs wont follow the aim assist right? i mean can you commit sometime later a rcs for silent aim too?
ps not native speaker sorry

**Comment by AntonDoil** [comment]
> hey i just wanna know is there no way RCS can be used with silent aim? I mean i use silent aim in my config but when i decide to spray i cant control the recoil. I know if rcs is coupled with silent aim rcs wont follow the aim assist right? i mean can you commit sometime later a rcs for silent aim too?
> ps not native speaker sorry

Ain't possible. 
When ur using silent aim you sending sendPacket, thats impossible to make RCS for silent.

**Comment by Asviix** [comment]
ah yes
finally I don't have to adapt the code everytime I update Osiris

**Comment by king1919djdj** [comment]
[ISSUE]
The rcs kicks in after firing 7-8 bullets, not from the starting.
even though i have set ignore shots to 0 or 1

**Comment by Gangster8192** [comment]
Makes csgo crash

**Comment by Gangster8192** [comment]
Broken when merging with the latest version of osiris.

**Comment by Asviix** [comment]
The game crashes when using the RCS with an automatic weapon, doesn't affect pistols tho

**Comment by spleatz** [comment]
Crashing when shooting and using random rcs. Debugged and this is the error:
https://media.discordapp.net/attachments/742586673552031785/770262793021816852/unknown.png?width=1204&height=677

**Comment by FalloutBruhMoments** [comment]
Is this still working?

**Comment by Asviix** [comment]
> Is this still working?

Only without Random RCS

**Comment by FalloutBruhMoments** [comment]
how to merge?

**Comment by linuxer9** [comment]
It's working rn without the random RCS, but it feels choppy when using it with the aimbot on; because the aimbot has its own RCS and it's set to the maximum value
so to resolve this problem, would you suggest disable recoil compensation while the standalone RCS is on ?
something like this maybe?

```
if (config->aimbot[weaponIndex].standaloneRCS)
{
    auto angle = calculateRelativeAngle(localPlayerEyePosition, bonePosition, cmd->viewangles);
}
else 
{
    auto angle = calculateRelativeAngle(localPlayerEyePosition, bonePosition, cmd->viewangles+aimPunch);
}
```
@r3muxd

**Comment by mkot2** [comment]
> 
> 
> It's working rn without the random RCS, but it feels choppy when using it with the aimbot on; because the aimbot has its own RCS and it's set to the maximum value
> so to resolve this problem, would you suggest disable recoil compensation while the standalone RCS is on ?
> something like this maybe?
> 
> ```
> if (config->aimbot[weaponIndex].standaloneRCS)
> {
>     auto angle = calculateRelativeAngle(localPlayerEyePosition, bonePosition, cmd->viewangles);
> }
> else 
> {
>     auto angle = calculateRelativeAngle(localPlayerEyePosition, bonePosition, cmd->viewangles+aimPunch);
> }
> ```
> 
> @r3muxd



> 
> 
> > Is this still working?
> 
> Only without Random RCS



> 
> 
> Crashing when shooting and using random rcs. Debugged and this is the error:
> https://media.discordapp.net/attachments/742586673552031785/770262793021816852/unknown.png?width=1204&height=677

the random float generator is faulty i suppose, using this generator random rcs works as well:
```
float floatGenerator(float bottom, float top) noexcept
{   
    std::random_device rd{};
    std::mt19937 generator(rd());
    std::uniform_real_distribution<float> dis(bottom, top);
    return dis(generator);
}
```

File: Osiris/Hacks/Aimbot.cpp [hunk 1]
```diff
@@ -97,6 +97,20 @@ static bool canScan(Entity* entity, const Vector& destination, const WeaponInfo*
...
 }
 
+static void setRandomSeed(int seed) noexcept
+{
+    using randomSeedFn = void(*)(int);
+    static auto randomSeed{ reinterpret_cast<randomSeedFn>(GetProcAddress(GetModuleHandleA("vstdlib.dll"), "RandomSeed")) };
+    randomSeed(seed);
+}
+
+static float getRandom(float min, float max) noexcept
```
Comment (mkot2) [inline Osiris/Hacks/Aimbot.cpp:107]: get rid of setRandomSeed and replace this with the code below, that way random rcs will work
```
float getRandom(float bottom, float top) noexcept
{   
    std::random_device rd{};
    std::mt19937 generator(rd());
    std::uniform_real_distribution<float> dis(bottom, top);
    return dis(generator);
}
```

**Review by mkot2 — CHANGES_REQUESTED** [review]

File: Osiris/Hacks/Aimbot.cpp (hunk 1 above)
Comment (linuxer9) [reply Osiris/Hacks/Aimbot.cpp:107]: thanks I will check it :)

**Comment by yabedabedu786** [comment]
update latest version?

**Review by WhatsupDanger — APPROVED** [review]

**Comment by kang-kung** [comment]
im getting a few errors, is it me or the code is outdated

'getRandom: redefinition; diffrent linkage
'getrandom': undeclared identifier
'*='unable to resolve function overload
'getrandom undeclared identifier
'*=' unable to resolve function overload

**Comment by zoominh0** [comment]
I mentioned above that I added a few lines of code and after I tried to build the solution, I got error c2737, at first I thought I fucked something up, but know I tried to build your normal rcs solution but I get the same error.

File: Osiris/Hacks/Aimbot.cpp (hunk 1 above)
Comment (zoominh0) [reply Osiris/Hacks/Aimbot.cpp:107]: Can you explain how to do this, what needs to be removed? when I add your code its gives me 100 error, admittedly I dont know what im doing lol.
Capture
Capture2

Comment (mkot2) [reply Osiris/Hacks/Aimbot.cpp:107]: you are missing #include <random>

Comment (zoominh0) [reply Osiris/Hacks/Aimbot.cpp:107]: I put that in aimbot settings?

Comment (zoominh0) [reply Osiris/Hacks/Aimbot.cpp:107]: And what do I do with setRandomseed, do I delete the set of lines for that code or just rename

Comment (mkot2) [reply Osiris/Hacks/Aimbot.cpp:107]: > 
> 
> **get rid of setRandomSeed** and replace this with the code below, that way random rcs will work
> 
> ```
> float getRandom(float bottom, float top) noexcept
> {   
>     std::random_device rd{};
>     std::mt19937 generator(rd());
>     std::uniform_real_distribution<float> dis(bottom, top);
>     return dis(generator);
> }
> ```

**Comment by Asviix** [comment]
Sometimes before actually spraying the view jumps up right from where I was aiming, is this a known bug ?

**Review by cheateronly123 — COMMENTED** [review]
I did everything the same as the code provided, but only errors came when compiling.

**Comment by tirziz** [comment]
seriously, this is so good why daniel didnt merge this, Its been 2 years lol

**Comment by fanat1cc** [comment]
Can you please update this as it doesn't work currently and messes up aimbot

**Comment by mrfuzz12** [comment]
Fix the dam aimbot wtf

**Pull request closed without merging**